src/dwc_checklist.R

```diff
@@ -201,32 +196,29 @@ distribution %<>% mutate(establishmentMeans = "introduced")
 
 #' #### appendixCITES
 #' #### eventDate
+#' 
+#' Distributions will have a start and end year, expressed in `eventDate` as ISO 8601 `yyyy/yyyy` (`start_year`/`end_year`). The `start_year` will contain the information from the first observation (`raw_first_occurrence_in_flanders`), which is expressed as `yyyy`, `< yyyy`, `<yyyy`, `before yyyy` or the first year of `yyyy-yyyy`. For the `end_year`, we will use the last year of `yyyy-yyyy` in `raw_first_occurrence_in_flanders` if available. If such an end year is not available, we have to assume when the presence of the species was last verified. We will use the publication year of Boets et al. 2016.
 #'
 #' Inspect content of `raw_first_occurrence_in_flanders`:
 distribution %>%
   distinct(raw_first_occurrence_in_flanders) %>%
   arrange(raw_first_occurrence_in_flanders) %>%
   kable()
 
-#' `eventDate` will be of format `start_year`/`end_year` (yyyy/yyyy).
-#' `start_year` (yyyy) will contain the information from the following formats in `raw_first_occurrence_in_flanders`: "yyyy", "< yyyy", "<yyyy" and "before yyyy" OR the first year of the interval "yyyy-yyyy":
-#' `end_year` (yyyy) is `2016` OR will contain the last year of the interval "yyyy-yyyy" in `raw_first_occurrence_in_flanders`:
 #' Before further processing, `raw_first_occurrence_in_flanders` needs to be cleaned, i.e. remove "<","< " and "before ":
 distribution %<>% mutate(year = str_replace_all(raw_first_occurrence_in_flanders, "(< |before |<)", ""))
 
 #' Create `start_year`:
-distribution %<>%
-  mutate(start_year = 
-           case_when(
-             str_detect(year, "-") == "TRUE" ~ "1730",   # when `year` = range --> pick first year (1730 in 1730-1732)
-             str_detect(year, "-") == "FALSE" ~ year))   
+distribution %<>% mutate(start_year = case_when(
+  str_detect(year, "-") == "TRUE" ~ strsplit(string,"-")[[1]][1], # when `year` = range --> pick first year (1730 in 1730-1732)
```

Comment:
The vector `string` is not specified in the script. However, even when I replace `string` by the vector `year`, the code does not seem to work properly, as it takes the first date in dataset as a value, rather than the first date in the date range.